controllers/utils/policy_template.go

```diff
@@ -0,0 +1,183 @@
+	// This regular expression is to extract all hub templates from a string.
+	re1 := regexp.MustCompile(`{{hub\s+.*?\s+hub}}`)
+
+	// This regular expression is to get the function name, resource name and namespace referenced in the function from a hub template.
+	// The following captured groups represent for:
+	// 		$1: any characters before the template function
+	// 		$2: template function
+	// 		$3: resource namespace field
+	// 		$4: resource namespace with printf variable
+	// 		$5: resource namespace with a fixed string
+	// 		$6: resource name field
+	// 		$7: resource name with printf variable
+	// 		$8: resource name with a fixed string
+	// 		$9: any characters after the resource name field
+	re2 := regexp.MustCompile(`({{hub.*)(fromConfigMap|fromSecret|lookup)\s+((\(\s*printf\s.+?\s*\))|"(.*?)")\s+((\(\s*printf\s.+?\s*\))|"(.*?)")(.*hub}})`)
+
+	var resolvedTemplates = templates
+	// Get all hub templates appeared in a string
+	discoveredTemplates := re1.FindAllString(templates, -1)
+	// Process each hub template
+	for _, template := range discoveredTemplates {
+		matches := re2.FindAllStringSubmatch(template, -1)
+
+		// Hub template doesn't match the regular expression
+		if len(matches) == 0 {
+			return "", &PolicyErr{template, PlcHubTmplFmtErr}
+		}
+
+		for _, match := range matches {
+			function := match[2]
+			if function != "fromConfigMap" {
+				return "", &PolicyErr{function, PlcHubTmplFuncErr}
+			}
+
+			if match[4] != "" {
+				return "", &PolicyErr{match[4], PlcHubTmplPrinfInNsErr}
+			}
+			namespace := match[5]
+
+			if match[7] != "" {
+				return "", &PolicyErr{match[7], PlcHubTmplPrinfInNameErr}
+			}
+			name := match[8]
+
+			fromNamespace := namespace
+			if namespace == "" {
+				// namespace is empty
+				fromNamespace = r.PolicyNamespace
+			}
+
+			fromResource := types.NamespacedName{
+				Name:      name,
+				Namespace: fromNamespace,
+			}
+
+			toResource := types.NamespacedName{
+				Name:      r.PolicyNamespace + "." + name,
+				Namespace: r.TargetNamespace,
+			}
+
+			if err := r.copyConfigmap(r.Ctx, fromResource, toResource); err != nil {
+				return "", err
+			}
+
+			// Update the hub templating with the replicated configmap name and namespace
+			updatedTemplate := re2.ReplaceAllString(template, `$1$2`+` "`+toResource.Namespace+`"`+` "`+toResource.Name+`"`+`$9`)
+			resolvedTemplates = strings.ReplaceAll(resolvedTemplates, template, updatedTemplate)
+		}
+	}
+
+	return resolvedTemplates, nil
+}
+
+//nolint:gocritic
+func (r *TemplateResolver) copyConfigmap(ctx context.Context, fromResource, toResource types.NamespacedName) error {
+	// Get the original configmap referenced in the inform policy
+	cm := &corev1.ConfigMap{}
+	err := r.Get(ctx, fromResource, cm)
+	if err != nil {
+		return err
+	}
+
+	copiedCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        toResource.Name,
+			Namespace:   toResource.Namespace,
+			Annotations: cm.GetAnnotations(),
```

Comment:
We should include here an indication that TALM copied the configmap. If I remember correctly multiple policies that refer to the same configmap will make only one copy (is that right?) and all of them may overwrite/update it at any time, so we can't actually have it "owned" by one CGU, but we should be able to at least note that TALM created it.

Reply:
Yes, we only have one copy for all policies. I added these labels to the copied CM. Do you think it's good enough? 

openshift-cluster-group-upgrades/fromCmName
openshift-cluster-group-upgrades/fromCmNamespace

Reply:
When we copy policies we annotate with:
`cluster-group-upgrades-operator/rname: .....`
For consistency I would say we should use the same prefix and combine into a single value (makes changes by multiple CGU atomic):
`cluster-group-upgrades-operator/fromCmName: namespace.name`

Reply:
Updated.